### PR TITLE
Feature/error handling

### PR DESCRIPTION
**Pull request by Codeherder19**
### What does this PR do?
- This PR displays error messages on initial page load as well as when you try to view an individual movie and it's details.
- This was tested by corrupting the file path in the API fetch request. We will test for 500 responses later.

### What does it fix?
- No bugs to fix.

### Is this a feature or a fix?
- This is a feature, providing an error message to the user.

### Where should the reviewer start?
- Clone this repo down to your machine. Run npm install then run npm start. 

### How should this be tested?
- While in MainDisplay.js, try messing up the file path for each one of the fetch calls respectively. Start with the get request for all movies and verify that an error message pops up. Then fix that get request and modify the get request for the individual movie call, try clicking on an individual movie and you should see an error message as well.

@letakeane
@sertmer

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -83,8 +82,7 @@ button {
   font-size: 1.5rem;
 }
 
-/* img {
-  object-fit: contain;
-  height: 350px;
-  width: auto;
-} */
+.error {
```
Comment (DrewBradley) [inline src/App.css:85]: Amazing styling choices! Beautiful!

File: src/components/MainDisplay.js [hunk 2]
```diff
@@ -54,7 +56,10 @@ class MainDisplay extends React.Component {
         date: data.movie.release_date
       }
     }))
-    console.log(this.state.currentMovie)
+    .catch(error => this.setState({
+      errorMessage: 'Oops! Something went wrong! Please try again later.',
+      movies: []
+    }))
```
Comment (DrewBradley) [inline src/components/MainDisplay.js:62]: This is a great response. They don't need to see the error code.

File: src/components/MainDisplay.js [hunk 3]
```diff
@@ -74,6 +79,11 @@ class MainDisplay extends React.Component {
       showIndividual={this.displayIndividual}/>)
     return (
       <div>
+        {this.state.errorMessage.length &&
+          <div>
+        <h1 className="error">{this.state.errorMessage}</h1>
+        <button onClick={this.componentDidMount}>BACK</button>
+          </div>}
```
Comment (DrewBradley) [inline src/components/MainDisplay.js:86]: Great idea to add a button so they can navigate back to the main page and try a different movie.

**Review by DrewBradley — APPROVED** [review]
Great work on this error handling, but what should happen if the request works, but no information is returned? Or if only partial information is returned?

File: src/App.css [hunk 4]
```diff
@@ -2,6 +2,7 @@ body {
   background-color: #000;
   width: 100%;
   height: 100vh;
+  display: flex;
```
Comment (letakeane) [inline src/App.css:5]: Yay, flexbox!

File: src/components/MainDisplay.js [hunk 5]
```diff
@@ -28,13 +28,15 @@ class MainDisplay extends React.Component {
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
     .then(response => response.json())
     .then(data => this.setState({
-      movies: data.movies
+      movies: data.movies,
+      errorMessage: ""
```
Comment (letakeane) [inline src/components/MainDisplay.js:32]: Good job updating two keys in state at once :)

File: src/components/MainDisplay.js [hunk 6]
```diff
@@ -28,13 +28,15 @@ class MainDisplay extends React.Component {
...
     }))
-  .catch(error => this.setState({errorMessage: error}))
+  .catch(error => this.setState({errorMessage: 'Oops! Something went wrong! Please try again later.'}))
```
Comment (letakeane) [inline src/components/MainDisplay.js:34]: Good customization of the error message for the end user

File: src/components/MainDisplay.js [hunk 7]
```diff
@@ -74,6 +79,11 @@ class MainDisplay extends React.Component {
       showIndividual={this.displayIndividual}/>)
     return (
       <div>
+        {this.state.errorMessage.length &&
+          <div>
+        <h1 className="error">{this.state.errorMessage}</h1>
```
Comment (letakeane) [inline src/components/MainDisplay.js:84]: Don't forget to make sure your indentation is consistent

File: src/components/Movie.js [hunk 8]
```diff
@@ -2,7 +2,6 @@ import React from 'react'
 
 const Movie = (props) => {
   return (
-    // {<div className="movie-thumb">}
       <div className="background-img"
```
Comment (letakeane) [inline src/components/Movie.js:5]: Good job getting rid of commented out code

**Review by letakeane — COMMENTED** [review]
Nice work; the error handling is concise! Good job with the conditional rendering.

**Comment by letakeane** [comment]
No need to make changes based on this because, of course, it's already merged in!

**Pull request merged**